### PR TITLE
Add pin pad to alarm panel

### PR DESCRIPTION
**Pull request by raccettura**
Using the UI from a tablet is overall enjoyable except the alarm more info panel is outright frustrating if you add a pincode... the native tablet keyboard is awkward when wall mounted requires setting focus.

This pull request (and associated PR for the backend) attempts to add a pin pad like most alarms use for when your code is nothing but integers. It's substantially easier to enable/disable this way.

This does need some UI cleanup, so hoping to get some assistance on that front. 

Based heavily on the ideas from @gazoscalvertos here:
https://community.home-assistant.io/t/yet-another-take-on-an-alarm-system/32386/1

The backend part of this PR is:
https://github.com/home-assistant/home-assistant/pull/14178

screen shot 2018-04-29 at 8 38 50 pm

### CODE REVIEW

**Comment by c727** [comment]
add a screenshot please

**Comment by raccettura** [comment]
Updated with screenshot

**Comment by c727** [comment]
I like this keypad design.

Are there any alarm panel that use none numeric codes? I'm not sure if we want an extra attribute for this

**Comment by raccettura** [comment]
Well my thinking was if the code is numeric show the keypad... if not use don't and use the existing UI. That would be more compatible with alarms that might use non-numeric codes or passwords... who knows how people are using the manual alarm. No reason to break existing functionality.

That said, this is still a rough cut.

File: src/dialogs/more-info/controls/more-info-alarm_control_panel.html [hunk 1]
```diff
@@ -9,8 +9,19 @@
 <dom-module id='more-info-alarm_control_panel'>
   <template>
     <style is="custom-style" include="iron-flex"></style>
+    <style>
+      #codepanel {
```
Comment (c727) [inline src/dialogs/more-info/controls/more-info-alarm_control_panel.html]: we prefer classes if we don't need the id. this code works better on mobile:
```css
.pad {
  display: flex;
  justify-content: center;
}
.pad div {
  display: flex;
  flex-direction: column;
}
```

File: src/dialogs/more-info/controls/more-info-alarm_control_panel.html [hunk 2]
```diff
@@ -20,6 +31,32 @@
         disabled='[[!codeInputEnabled]]'
       ></paper-input>
     </div>
+
```
Comment (c727) [inline src/dialogs/more-info/controls/more-info-alarm_control_panel.html]:
```html
<template is="dom-if" if="[[numericPin]]">
  <div class="pad">
    <div>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="1" raised>1</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="4" raised>4</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="7" raised>7</paper-button>
    </div>
    <div>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="2" raised>2</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="5" raised>5</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="8" raised>8</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="0" raised>0</paper-button>
    </div>
    <div>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="3" raised>3</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="6" raised>6</paper-button>
      <paper-button on-click='numberPadClicked' disabled='[[!codeInputEnabled]]' data-digit="9" raised>9</paper-button>
      <paper-button on-click='callclearcode' disabled='[[!codeInputEnabled]]' raised>Clear</paper-button>
    </div>
  </div>
</template>
```

**Comment by hawk259** [comment]
👍 Great work!

Could we get a send button and # key added? 

Right now there is no way to bypass zones, with a psychical key pad I can do `<user code>6#` to bypass all faulted zones or `<user code>6<zone>[<zone>...]` to bypass a single/multiple specific zones.

**Comment by raccettura** [comment]
I think ultimately it might make sense to make arm/disarm a button on the keypad... action flips depending on the current state (says 'disarm' when it's armed, 'arm' when disarmed).  Then any additional options go below where the current buttons are. That would cover the most common use case on the keypad but still allow for other alarm states to be implemented. Might also make sense to have some color indicator to make it easier to detect the current state, something that this dialog currently handles very poorly.

**Review by c727 — APPROVED** [review]

**Comment by syssi** [comment]
@raccettura Please resolve the merge conflict.

**Comment by raccettura** [comment]
Updated pull request. app.js doesn't seem to be building for me, and I'm out of time to dig into why that is. However I think this PR should be in decent shape.

**Comment by raccettura** [comment]
Anything else needed here?

**Comment by c727** [comment]
Looks nice :+1: 

now someone has to pimp the arm-buttons a bit; maybe only 1 centered button each line...

**Pull request merged**